service/matching/task_queue_partition_manager.go

```diff
@@ -275,15 +278,28 @@ func (pm *taskQueuePartitionManagerImpl) PollTask(
 		// update timestamp when long poll ends
 		defer pm.defaultQueue.UpdatePollerInfo(pollerIdentity(identity), pollMetadata)
 	}
-	return dbq.PollTask(ctx, pollMetadata)
+
+	task, err := dbq.PollTask(ctx, pollMetadata)
+	return task, versionSetUsed, err
 }
 
 func (pm *taskQueuePartitionManagerImpl) DispatchSpooledTask(
 	ctx context.Context,
 	task *internalTask,
+	sourceBuildId string,
 ) error {
+	if sourceBuildId != "" {
+		// Task is spooled in a build ID queue. it should be dispatched to the same build ID until
+		// we add redirect rules support.
+		pq, err := pm.getVersionedQueue(ctx, "", sourceBuildId, true)
+		if err != nil {
+			return err
+		}
+		return pq.DispatchSpooledTask(ctx, task, nil)
+	}
 	taskInfo := task.event.GetData()
 	// This task came from taskReader so task.event is always set here.
+	// TODO: in WV2 we should not look at a spooled task directive anymore [cleanup-old-wv]
```

Comment:
btw, if we don't use the directive in TaskInfo anymore, can we remove it from there?

Reply:
sure, I marked the field as deprecated to be deleted with old WV cleanup.